unit-tests/live/rec-play/test-record-software-device.py

```diff
@@ -0,0 +1,143 @@
+# License: Apache 2.0. See LICENSE file in root directory.
+# Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+import pyrealsense2 as rs
+from rspy import test
+import numpy as np
+import tempfile, os.path
+
+################################################################################################
+test.start("Record software-device")
+
+W = 640
+H = 480
+BPP = 2
+
+temp_dir = tempfile.mkdtemp()
+filename = os.path.join(temp_dir, "recording.bag")
+
+video_frame = rs.software_video_frame()
+motion_frame = rs.software_motion_frame()
+
+pixels = np.array([100 for i in range(W*H*BPP)], dtype=np.uint8)
+motion_frame_data = rs.vector()
+motion_frame_data.x = 1.0
+motion_frame_data.y = 2.0
+motion_frame_data.z = 3.0
+
+sd = rs.software_device()
+sensor = sd.add_sensor("Synthetic")
+
+depth_intrinsics = rs.intrinsics()
+depth_intrinsics.width = W
+depth_intrinsics.height = H
+depth_intrinsics.ppx = float(W/2)
+depth_intrinsics.ppy = H/2
```

Comment:
ppx and ppy are same type.
How come only one need float casting and the other not?